_posts/2019-08-09-automated-prize-draws-kafka.md

```diff
@@ -0,0 +1,38 @@
+---
+layout:     post
+title:      Automated Prize Draws with Kafka
+date:       2019-08-09
+summary:    "A look at how we improve on our existing manual processes and remove customer pain using technology."
+author:     dave_chaston
+tags:       kafka, kstreams, gaming, promotions
+category:   Software Engineering
+---
+
+In the Next Generation Promotions (NGP) squad, we're always looking for new ways to be the market leader for offering great promotions. This could mean faster ways of rewarding (Choose your Cashback Hour meaning players get rewarded at the end of their time slot, rather than waiting for some overnight process to finish), better rewards (Different types of freespins, available across different game providers), or even new and improved mechanics. This post details how we've taken our old prize draw mechanic and saved our CRM and Gaming Operations teams time and effort to run them, automating something which previously would have taken hours of manual effort to figure out which customer won what and then get the prizes out to them. It also gives us the ability to run them at any time, rather than them having to start and end at midnight, and lets us give customers feedback instantly on their current ticket counts.
+
+### The Problem:
+A prize draw campaign takes up a lot of effort from multiple people in order to be run, and as such doesn't tend to be utilised as much as it could be. In addition to this the feedback from customers was negative about the use of prize draws, with many either not knowing how many tickets they have before the draw, or not really believing that prizes are awarded at all as they never hear about the results when they lose. This has a further effect on our Customer Experience team as they regularly receive calls from customers who want to know the status of the promotion and how many tickets they currently have.
+
+### The Process:
+The current main use case for prize draws is that we award a number of tickets (usually either 1 or 5 per person) as a lower tier Prize Machine prize to customers who win that day. If the customer opts in, these tickets are collated at draw time, run through Unica, and selected at random for who wins what. The most common example for Prize Machine is that 200 people will win some free spins. This can also be done based on stake if needed, with the number of tickets determined at draw time based on the amount staked by each customer.
```

Comment:
I had to google what "Unica" is. Maybe worth linking to something, or adding a footnote, to explain what that is for people unfamiliar.

Similarly for Prize Machine. Sky Vegas users know what that is, but a short explanation for people unfamiliar may be useful. :)